_includes/head.html

```diff
@@ -23,10 +23,16 @@
     </title>
     <meta name="description" content="{{ site.description }}">
 
+    {% if page.canonical %}
+        {% assign canonical_url = page.canonical | replace:'index.html','' | prepend: site.url %}
```

Comment:
I don't think it's actually necessary, since ```index.html``` will probably only appear in page.url, whereas page.canonical is built up in the .rb without that?